roles/internal/crayfish/tasks/install.yml

```diff
@@ -0,0 +1,87 @@
+---
+
+- name: Install requisite package repositories
+  apt_repository:
+    repo: 'ppa:lyrasis/imagemagick-jp2'
+
+- name: Install requisite packages
+  apt:
+    name: "{{item}}"
+    state: present
+  with_items:
+    - tesseract-ocr
+    - imagemagick
+
+- name: Install crayfish code
+  git:
+    repo: https://github.com/Islandora-CLAW/Crayfish.git
+    dest: "/var/www/html/Crayfish"
+    version: "{{ crayfish_version_tag }}"
+    
+- name: Build crayfish code including dependencies
+  composer:
+    command: install
+    working_dir: /var/www/html/Crayfish/{{item}}
+  with_items: "{{ crayfish_services }}"
+    
+- name: Configure crayfish code
+  template:
+    src: "{{item}}.config.yaml.j2"
+    dest: "/var/www/html/Crayfish/{{item}}/cfg/config.yaml"
+  with_items: "{{ crayfish_services }}"
+    
+- name: Install auth config
+  template:
+    src: "syn-settings.xml.jp2"
+    dest: "/var/www/html/Crayfish/{{item}}/syn-settings.xml"
+  with_items: "{{ crayfish_services }}"
+
+- name: Create Gemini db
+  mysql_db:
+    name: gemini
+    state: present
+
+- name: Grab Gemini db schema
+  template:
+    src: "gemini.sql"
+    dest: "/tmp/gemini.sql"
+
+- name: Install Gemini db schema
+  mysql_db:
```

Comment:
So this is idempotent, because it has the `IF NOT EXISTS` in the sql. However, its idiomatic in ansible to keep ansible from reporting `changed` on idempotent tasks, so if you are running a bunch of tasks, its easy to see what changed. If you run this task twice, it will always report changed right now.

I think I would rewrite this section this way to keep it from reporting changed: 
```yaml
- name: Create Gemini db
  mysql_db:
    name: "{{ gemini_db_options.dbname }}"
    state: present
  register: gemini_db_exists

- name: Grab Gemini db schema
  template:
    src: "gemini.sql"
    dest: "/tmp/gemini.sql"
  when: gemini_db_exists.changed

- name: Install Gemini db schema
  mysql_db:
    state: import
    name: all
    target: "/tmp/gemini.sql"
  when: gemini_db_exists.changed
```

That way we will only update the schema when the table is first created, and ansible won't report that the tasks have changed.

Reply:
Ah, I didn't know about `something.changed`. Is that a universal thing in Ansible?

Reply:
Yup! its super handy for things like this sometimes.